app/browser/ads/adBlockUtil.js

```diff
@@ -31,16 +31,25 @@ const mapFilterType = {
 const shouldDoAdBlockCheck = (resourceType, firstPartyUrl, url, shouldCheckMainFrame) =>
   firstPartyUrl.protocol &&
     // By default first party hosts are allowed, but enable the check if a flag is specified in siteHacks
+  (
     shouldCheckMainFrame ||
-    (resourceType !== 'mainFrame' &&
-      isThirdPartyHost(firstPartyUrl.hostname || '', url.hostname) ||
-       siteHacks[firstPartyUrl.hostname] && siteHacks[firstPartyUrl.hostname].allowFirstPartyAdblockChecks) &&
-    // Only check http and https for now
-    firstPartyUrl.protocol.startsWith('http') &&
-    // Only do adblock if the host isn't in the whitelist
-    !whitelistHosts.find((whitelistHost) => whitelistHost === url.hostname || url.hostname.endsWith('.' + whitelistHost)) &&
-    // Make sure there's a valid resource type before trying to use adblock
-    mapFilterType[resourceType] !== undefined
+    (
+      resourceType !== 'mainFrame' &&
+      (
+        isThirdPartyHost(firstPartyUrl.hostname || '', url.hostname) ||
+        (
+          siteHacks[firstPartyUrl.hostname] &&
+          siteHacks[firstPartyUrl.hostname].allowFirstPartyAdblockChecks
+        )
+      ) &&
```

Comment:
I think this `)` should be in L39 so that `resourceType !== 'mainFrame'` only affect `isThirdPartyHost(firstPartyUrl.hostname || '', url.hostname)`